sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedCommands.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, ShowColumns, ShowNamespaces, ShowPartitions, ShowTableProperties, ShowTables}
+import org.apache.spark.sql.catalyst.rules.Rule
+
+/**
+ * Rule that resolves table-valued command references.
+ */
+object ResolveTableValuedCommands extends Rule[LogicalPlan] {
+
+  private def isSupportedCommand(plan: LogicalPlan): Boolean = plan match {
+    case ShowNamespaces(_, _, _) => true
+    case ShowTables(_, _, _) => true
+    case ShowTableProperties(_, _, _) => true
+    case ShowPartitions(_, _, _) => true
+    case ShowColumns(_, _, _) => true
+    // TODO   case ShowViews(_, _, _) => true
```

Comment:
Can we file a JIRA and make it id'ed todo? e.g) `TODO(SPARK-XXXX): blah blah`

Reply:
If https://github.com/apache/spark/pull/31508 and https://github.com/apache/spark/pull/31519 before this PR merged, I will implement the code. Otherwise, I will create two tickets later.

Reply:
👌